### ISSUE TITLE
Ordering is hard to control when optimizing multiple files

### ISSUE THREAD

**Issue by Grandrath** [jakubpawlowicz/clean-css#908]
When optimizing multiple files from a hash the result is depending on the ordering of the keys:
```javascript
new CleanCSS().minify({
  "a.css": {styles: "body{color:red;}"},
  "b.css": {styles: "body{color:green;}"}
}).styles
// => 'body{color:red}body{color:green}'

new CleanCSS().minify({
  "b.css": {styles: "body{color:green;}"},
  "a.css": {styles: "body{color:red;}"}
}).styles
// => 'body{color:green}body{color:red}'
```
The reason is the `for...in` loop in the [`fromHash`](https://github.com/jakubpawlowicz/clean-css/blob/master/lib/reader/read-sources.js#L71) function. I am aware that the ES6 standard [specifies an iteration order](http://www.2ality.com/2015/10/property-traversal-order-es6.html). However this order is "in the order in which [the keys] were added to the object" and not, say, alphabetically. This makes it unnecessarily hard to construct the input hash for the desired output result.

I propose changing the signature for multiple files to take an array instead of a hash:
```javascript
new CleanCSS().minify([
  {"a.css": {styles: "body{color:red;}"}},
  {"b.css": {styles: "body{color:green;}"}}
])
```
This way the ordering is obvious and easily controllable by the caller.

I willing to provide a PR in case you are interested in this change.

**Comment by jakubpawlowicz** [comment]
@Grandrath thanks for bringing that up

I wouldn't mind adding another signature beside ones we already have, i.e. passing an array or a hash.
I think we should leave existing ones as is since it'd be a breaking change.

PR are more than welcome - https://github.com/jakubpawlowicz/clean-css/blob/master/lib/reader/read-sources.js#L34 is a good place to start

**Comment by Grandrath** [comment]
> I wouldn't mind adding another signature beside ones we already have, i.e. passing an array or a hash.
I think we should leave existing ones as is since it'd be a breaking change.

I agree. There is no reason for breaking existing code.

> PR are more than welcome

I'm on it. :-)

**Comment by jakubpawlowicz** [comment]
Fixed by #918, thanks for the patch @Grandrath!
